core/data_operations/account_data.py

```diff
@@ -28,14 +32,65 @@ def check_account_exists_then_return(
             return {
                 "account_id": account.id,
                 "email_address": account.email,
-            }
+                "roles": account.serialize["roles"],
+            }, 200
         else:
             return account
     except sqlalchemy.exc.NoResultFound:
         return NoContent, 404
 
 
-def get_account_data_by_email(email: str, as_json: bool = True) -> Tuple[dict, int]:
+def update_account(account_id: str, roles: List[str]) -> Tuple[dict, int]:
+    """
+    Updates an account rols
+    Args:
+        account_id (str): The account id
+        roles (List[str]): A list of roles to set
+
+    Returns:
+        A tuple with account status as json and a status code.
+
+    """
+    # Check account exists
+    try:
+        account = (
+            db.session.query(Account).filter(Account.id == account_id).one()
+        )
+    except sqlalchemy.exc.NoResultFound:
+        return NoContent, 404
+    # Check all roles are valid
+    for role in roles:
+        try:
+            RoleType[role.upper()]
+        except KeyError:
+            return {"error": f"Role '{role}' is not valid"}, 401
+    # Delete existing roles
+    stmnt = delete(Role).where(Role.account_id == account_id)
+    db.session.execute(stmnt)
+
+    # Update current roles
+    current_roles = []
+    for role in roles:
+        current_role = Role()
+        current_role.account_id = account_id
+        current_role.role = RoleType[role.upper()]
```

Comment:
⛏️ Postgres only saves the "left hand side" of your enums - ie the attribute names. Look in the migrations for proof (line https://github.com/communitiesuk/funding-service-design-account-store/blob/fe520411d760f37d37f9440be24a787128db0618/db/migrations/versions/d197951abf53_.py#L39 ). So you don't need to use `RoleType` to convert here. You can just pass the capitalised role.

Reply:
Obv